src/messaging/ExchangeContext.cpp

```diff
@@ -300,6 +300,21 @@ bool ExchangeContext::MatchExchange(SecureSessionHandle session, const PacketHea
         && (payloadHeader.IsInitiator() != IsInitiator());
 }
 
+void ExchangeContext::OnConnectionExpired()
+{
+    if (!IsResponseExpected())
+    {
+        // Nothing to do in this case
+        return;
+    }
+
+    // If we're waiting on a response, we now know it's never going to show up
+    // and we should notify our delegate accordingly.
+    CancelResponseTimer();
+    SetResponseExpected(false);
+    NotifyResponseTimeout();
+}
```

Comment:
ExchangeContext is created with a valid SecureSession, is mSecureSession still valid after the associated connection get closed?

Reply:
That is a good question.  None of that stuff is documented, so I don't know.  @pan-apple ?

Reply:
Per discussion with @pan-apple we should reset `mSecureSession` here so no more messages can be sent on this exchange.